src/Fixer/Operator/TernaryToNullCoalescingFixer.php

```diff
@@ -29,59 +29,11 @@
      */
     public function fix(\SplFileInfo $file, Tokens $tokens)
     {
-        $issets = array_keys($tokens->findGivenKind(T_ISSET));
-
-        while ($issetIndex = array_pop($issets)) {
-            $prevTokenIndex = $tokens->getPrevMeaningfulToken($issetIndex);
-            if ($this->isHigherPrecedenceAssociativityOperator($tokens[$prevTokenIndex])) {
-                continue;
-            }
-
-            $startBraceIndex = $tokens->getNextTokenOfKind($issetIndex, array('('));
-            $endBraceIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $startBraceIndex);
-
-            $ternaryQuestionMarkIndex = $tokens->getNextMeaningfulToken($endBraceIndex);
-            if (!$tokens[$ternaryQuestionMarkIndex]->equals('?')) {
-                // we are not in a ternary operator
-                continue;
-            }
-
-            // search what is inside the isset()
-            $issetTokens = $this->getMeaningfulSequence($tokens, $startBraceIndex, $endBraceIndex);
-
-            // search what is inside the middle argument of ternary operator
-            $ternaryColonIndex = $tokens->getNextTokenOfKind($ternaryQuestionMarkIndex, array(':'));
-            $ternaryFirstOperandTokens = $this->getMeaningfulSequence($tokens, $ternaryQuestionMarkIndex, $ternaryColonIndex);
-
-            if ($issetTokens->generateCode() !== $ternaryFirstOperandTokens->generateCode()) {
-                // regardless of non-meaningful tokens, the operands are different
-                continue;
-            }
-
-            if ($this->hasChangingContent($issetTokens)) {
-                // some weird stuff inside the isset
-                continue;
+        // only analyze tokens between minimal number of tokens needed for a candidate.
+        for ($i = count($tokens) - 9; $i > 2; --$i) {
```

Comment:
Here I see the performance improvement, but I think the gain is less valuable than the delegation of correctness and maintenance to the `Tokens::findGivenKind`